pkg/receive/multitsdb.go

```diff
@@ -595,7 +605,17 @@ func (t *MultiTSDB) RemoveLockFilesIfAny() error {
 
 func (t *MultiTSDB) TSDBLocalClients() []store.Client {
 	t.mtx.RLock()
-	defer t.mtx.RUnlock()
+	if !t.tsdbClientsNeedUpdate {
+		t.mtx.RUnlock()
+		return t.tsdbClients
+	}
+
+	t.mtx.RUnlock()
+	t.mtx.Lock()
+	if !t.tsdbClientsNeedUpdate {
+		return t.tsdbClients
+	}
+	defer t.mtx.Unlock()
```

Comment:
`defer t.mtx.Unlock()` should be before `if`. @jnyi

Reply:
Thanks; care opening up a PR with the fix?

Reply:
Already fixed here https://github.com/thanos-io/thanos/commit/c10b695a5ba34f05e2328d74dd5e48159137b11e

Reply:
no, we found a different bug, I submit a PR to fix it, please take a look when you have time: https://github.com/thanos-io/thanos/pull/7941